// react-native-agora/lib/commonjs/internal/event.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.EVENT_TYPE = exports.DeviceEventEmitter = void 0;
exports.emitEvent = emitEvent;
var _eventemitter = _interopRequireDefault(require("eventemitter3"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
// @ts-ignore
const DeviceEventEmitter = exports.DeviceEventEmitter = new _eventemitter.default();

/**
 * @internal
 */
let EVENT_TYPE = exports.EVENT_TYPE = /*#__PURE__*/function (EVENT_TYPE) {
  EVENT_TYPE[EVENT_TYPE["IMediaEngine"] = 0] = "IMediaEngine";
  EVENT_TYPE[EVENT_TYPE["IMediaPlayer"] = 1] = "IMediaPlayer";
  EVENT_TYPE[EVENT_TYPE["IMediaRecorder"] = 2] = "IMediaRecorder";
  EVENT_TYPE[EVENT_TYPE["IRtcEngine"] = 3] = "IRtcEngine";
  EVENT_TYPE[EVENT_TYPE["IMusicContentCenter"] = 4] = "IMusicContentCenter";
  EVENT_TYPE[EVENT_TYPE["IAgoraH265Transcoder"] = 5] = "IAgoraH265Transcoder";
  EVENT_TYPE[EVENT_TYPE["IAgoraPip"] = 6] = "IAgoraPip";
  return EVENT_TYPE;
}({});
/**
 * @internal
 */
function emitEvent(eventType, eventProcessor, data) {
  DeviceEventEmitter.emit(eventType, eventProcessor, data);
}
